### PR TITLE
node incoming outgoing degree

### PR DESCRIPTION
**Pull request by swilly22**
#593

### CODE REVIEW

File: docs/commands.md [hunk 1]
```diff
@@ -456,6 +456,12 @@ This section contains information on all supported functions from the Cypher que
 | ------- |:-----------|
 |exists() | Returns true if the specified property exists in the node or relationship. |
 
+## Node functions
+|Function | Description|
+| ------- |:-----------|
+|indegree() | Returns the number of node's incoming edges. |
+|outdegree() | Returns the number of node's outgoing edges. |
```
Comment (jeffreylovitz) [inline docs/commands.md]: We should add some documentation on how these functions can optionally filter by relationship type.

File: src/arithmetic/arithmetic_expression.c [hunk 2]
```diff
@@ -912,6 +879,45 @@ SIValue AR_CASEWHEN(SIValue *argv, int argc) {
...
+	return res;
+}
+
+SIValue AR_INCOMEDEGREE(SIValue *argv, int argc) {
+	return _AR_NodeDegree(argv, argc, GRAPH_EDGE_DIR_INCOMING);
+}
+
+SIValue AR_OUTGOINGDEGREE(SIValue *argv, int argc) {
+	return _AR_NodeDegree(argv, argc, GRAPH_EDGE_DIR_OUTGOING);
+}
```
Comment (jeffreylovitz) [inline src/arithmetic/arithmetic_expression.c]: We can optionally also add a `degree` function for `GRAPH_EDGE_DIR_BOTH`, but it's not strictly necessary.

File: docs/commands.md (hunk 1 above)
Comment (swilly22) [reply docs/commands.md]: Agree, I think each function on this list should get its own section
@K-Jo is there any chance of us getting help extending our documentation from probably Ben?

File: src/arithmetic/arithmetic_expression.c (hunk 2 above)
Comment (swilly22) [reply src/arithmetic/arithmetic_expression.c]: I guess if we were to add a simple degree(n) function, then BOTH can be use.

**Pull request merged**